libraries/chain/wasm_interface.cpp

```diff
@@ -689,8 +689,8 @@ class console_api : public context_aware_api {
    public:
       using context_aware_api::context_aware_api;
 
-      void prints(const char *str) {
-         context.console_append(str);
+      void prints(null_terminated_ptr str) {
+         context.console_append((const char*)str);
```

Comment:
halp! How can I get rid of this cast, I'm missing some template magic.......

Reply:
`console_append` is a template and therefore requires an exact match on the argument. You could do this: `context.console_append<const char*>(str);` which is much better than the cast. Even better in my opinion would be to add a `.get()` to `null_terminated_ptr`. I am not a fan of `operator U*()`;

Reply:
I agree with @heifner about `context.console_append<const char*>(str);` being the best option at the moment.  

If we want to get crazy, we can use SFINAE to remove all the overloads for `console_append` which don't have a defined `operator <<` for the `std::ostringstream`.  This would remove the templated overload that accepts `null_terminated_ptr` and it looks like the compiler is smart enough to call the cast operator at that point.

at least according to this quickie I put together : https://godbolt.org/g/5LdbFB

Reply:
I want to be clear, the only reason we would do this insanity is if there is a high chance we will add more things like this. explicit template is pretty much ok 😄